ols/app/models/config.py

```diff
@@ -5,580 +5,213 @@
 from typing import Optional
 from urllib.parse import urlparse
 
-from pydantic import BaseModel
+from pydantic import BaseModel, field_validator, PositiveInt
+from typing import Literal
 
 from ols import constants
 
 
+# Helper functions
 def _is_valid_http_url(url: str) -> bool:
```

Comment:
Pydantic has a [validator](https://docs.pydantic.dev/latest/api/networks/) for URLs.